### PR TITLE
League

### PR DESCRIPTION
**Pull request by dekellevy93**
after adding hashmaps of referees to League And Season

### CODE REVIEW

File: src/main/java/com/SAS/League/League.java [hunk 1]
```diff
@@ -150,9 +154,28 @@ public void addPointsPolicy(Season season, PointsPolicy pointsPolicy) {
...
+    public HashMap<Season,HashSet<Referee>> getReferees() {
+        return referees;
+    }
+
+    /**
+     * @param season     the season that the  referee is relevant to
+     * @param ref the referee you want to add to the list of referees
+     */
+    public void addReferee(Season season, Referee ref) {
+        if (this.seasonList.contains(season) && !this.referees.containsKey(season)) {
```
Comment (AmitNachimovitz) [inline src/main/java/com/SAS/League/League.java]: the condition should be positive for this.referees.containsKey(season) no?
I think you want the referees list to contain the season

File: src/main/java/com/SAS/League/Season.java [hunk 2]
```diff
@@ -216,4 +210,24 @@ public void addRankPolicy(League league, LeagueRankPolicy rankPolicy) {
...
+        return referees;
+    }
+
+
+    /**
+     * @param league     the league that the  referee is relevant to
+     * @param ref the referee you want to add to the list of referees
+     */
+    public void addReferee(League league, Referee ref) {
+        if (this.leaguesList.contains(league)&&!this.rankPolicy.containsKey(league)) {
```
Comment (AmitNachimovitz) [inline src/main/java/com/SAS/League/Season.java]: why do you need to check in rank policy?

File: src/main/java/com/SAS/League/League.java (hunk 1 above)
Comment (dekellevy93) [reply src/main/java/com/SAS/League/League.java]: you're right, must have slipped from a a different function

File: src/main/java/com/SAS/League/Season.java (hunk 2 above)
Comment (dekellevy93) [reply src/main/java/com/SAS/League/Season.java]: i need to verify it doesn't have a rank policy yet.

File: src/main/java/com/SAS/League/Budget.java [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Map;
 
 public class Budget {
     private Team team;
     private Season season;
-    private int budget;
+    private double budget;
```
Comment (yaarm) [inline src/main/java/com/SAS/League/Budget.java]: Maybe change the name of the variable to amount.

File: src/main/java/com/SAS/League/Season.java [hunk 4]
```diff
@@ -43,7 +37,8 @@ public Season(int year, HashSet<Team> teamsList, HashSet<League> leaguesList) {
         this.gamesList = new HashMap<>();
         this.teamsList = teamsList;
         this.leaguesList = leaguesList;
-        this.budgets = new HashMap<>();
+        this.budgets = new HashMap<Team, Budget>();
+        this.referees= new HashMap<Referee, Budget>();
```
Comment (yaarm) [inline src/main/java/com/SAS/League/Season.java]: Why does the referees has budget??

File: src/main/java/com/SAS/League/Season.java (hunk 2 above)
Comment (yaarm) [reply src/main/java/com/SAS/League/Season.java]: Why is it relates to the adding of the new referee?

File: src/main/java/com/SAS/League/Season.java (hunk 4 above)
Comment (dekellevy93) [reply src/main/java/com/SAS/League/Season.java]: my bad. fixed it

File: src/main/java/com/SAS/League/Budget.java (hunk 3 above)
Comment (dekellevy93) [reply src/main/java/com/SAS/League/Budget.java]: changed

**Review by yaarm — APPROVED** [review]

**Review by yaarm — COMMENTED** [review]
Only resolve the conflicts first.

**Pull request merged**